amp/WEB-INF/src/org/digijava/module/aim/util/ActivityVersionUtil.java

```diff
@@ -666,24 +681,20 @@ private static void processVersionableSimple(AmpActivityVersion ampActivityOne,
         }
     }
 
-    public static Map<String, Map<String, List<CompareOutput>>> getOutputCollectionGrouped() {
+    public static Map<String, Map<String, List<CompareOutput>>> getOutputCollectionGrouped() throws Exception {
 
-        Map<String, Map<String, List<CompareOutput>>> listOfOutputCollectionGrouped = new HashMap<>();
+    	List<Object[]> lists = AuditLoggerUtil.getListOfActivitiesFromAuditLogger();
 
-        //Use lambda through the accept method from java consumer functional interface
-        // to create listOfOutputCollectionGrouped
-        AuditLoggerUtil.getListOfActivitiesFromAuditLogger().forEach((Object[] activityObj) -> {
-            Map<String, List<CompareOutput>> compareOutput;
-            try {
-                compareOutput = compareActivities(Long.parseLong(String.valueOf(activityObj[0]).trim()));
-                if (compareOutput != null) {
-                    listOfOutputCollectionGrouped.put(String.valueOf(activityObj[1]).trim(), compareOutput);
-                }
-            } catch (Throwable e) {
-                return; //Get rid if any exception in the current iteration and continue with next iteration
-            }
+        //Since this is short-lived async task
+        CountDownLatch signal = new CountDownLatch(lists.size());
```

Comment:
even a RunnableTask is not the proper Approach for that I stated below, when working with parallel tasks we need to have control of the amount of records we will run in parallel if we run all the records it will probably end up running slower than running one by one. I will work tomorrow in my approach using parallel streams and CompletableFuture from the Java 8 Stream api and will schedule a call for next Wednesday

Reply:
That's why I set lists.size() in CountDownLatch.